pkg/provisioner/ironic/ironic.go

```diff
@@ -1219,6 +1220,33 @@ func (p *ironicProvisioner) Delete() (result provisioner.Result, err error) {
 		return p.setMaintenanceFlag(ironicNode, true)
 	}
 
+
+	p.log.Info("requesting ports to delete")
+
+	err = ports.List(p.client, ports.ListOpts{NodeUUID: p.status.ID}).EachPage(func(page pagination.Page) (provisioner.Result, error) {
+		port_list, err := ports.ExtractPorts(page)
+		if err != nil {
+			return result, errors.Wrap(err, "failed to extract port list")
+		}
+		for i := 0; i < len(port_list) ; i++ {
+			p.log.Info("starting delete of port",
+				"ID", port_list[i].UUID,
+			) 
+			err = ports.Delete(p.client, port_list[i].UUID).ExtractErr()	
+			switch err.(type) {
+			case nil:
+				p.log.Info("port removed")
+			case gophercloud.ErrDefault409:
+				return result, errors.Wrap(err, "failed to remove as it is presently locked")
```

Comment:
Do we need to set `result.Dirty = true` so we try again?

Reply:
Looking further down, it looks like result.Dirty is only set to true upon success, but I was wondering the same thing.

Reply:
Returning an error causes a retry.